File: Compiladores(ano passado)/compact12b/compact/java/NodeBinary.java
import java.util.*;

public class NodeBinary extends NodeUnary {
	protected Node arg;
	NodeBinary(Node arg1, Node arg2) { super(arg1); arg = arg2; }
	NodeBinary(Node arg1, Node arg2, int lineno) {
		super(arg1, lineno); arg = arg2;
	}
	public Node second() { return arg; }
	public void accept(Compiler c) { c.nodeBinary(this); }
}

File: Compiladores(ano passado)/compact12b/compact/c++/NodeBinary.cpp
#include "NodeBinary.hpp"

void NodeBinary::accept(Compiler& c) { c.nodeBinary(*this); }
NodeBinary::~NodeBinary() { delete arg; }
void NodeBinary::print(std::ostream *out) {
	if (out) (*out) << "NodeBinary: ";
	NodeUnary::print(out);
	arg->print(out);
}
